parquet-column/src/main/java/org/apache/parquet/column/values/deltastrings/DeltaByteArrayWriter.java

```diff
@@ -82,11 +82,11 @@ public String memUsageString(String prefix) {
   @Override
   public void writeBytes(Binary v) {
     int i = 0;
-    byte[] vb = v.getBytes();
+    byte[] vb = v.getBytesUnsafe();
     int length = previous.length < vb.length ? previous.length : vb.length;
     for(i = 0; (i < length) && (previous[i] == vb[i]); i++);
     prefixLengthWriter.writeInteger(i);
-    suffixWriter.writeBytes(Binary.fromByteArray(vb, i, vb.length - i));
+    suffixWriter.writeBytes(Binary.fromUnmodifiedByteArray(vb, i, vb.length - i));
```

Comment:
The contract for getBytesUnsafe is to use them immediately so it doesn't matter if they change. If you pass them on, then that's not necessarily an immediate use. The consumer certainly shouldn't be changing the bytes in a Binary, but we mostly care about the consumer's intended use.